dotCMS/src/main/java/com/dotmarketing/services/ContentletServices.java

```diff
@@ -168,6 +171,10 @@ private static void invalidate(Contentlet content, Identifier identifier, boolea
 			//remove from cache
 			VanityUrlServices.getInstance().invalidateVanityUrl(content);
 		}
+		if(content != null && content.isKeyValue()){
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)